test/e2e/azure_test.go

```diff
@@ -597,4 +597,55 @@ var _ = Describe("Workload cluster creation", func() {
 			})
 		})
 	})
+
+	Context("Creating clusters using clusterclass [REQUIRED]", func() {
+		It("with a single control plane node, one linux worker node, and one windows worker node", func() {
+			// use "cc" as spec name because natgw pip name exceeds limit.
+			clusterName = getClusterName(clusterNamePrefix, "cc")
+
+			// Opt into using windows with prow template
+			Expect(os.Setenv("WINDOWS_WORKER_MACHINE_COUNT", "1")).To(Succeed())
+			Expect(os.Setenv("K8S_FEATURE_GATES", "WindowsHostProcessContainers=true")).To(Succeed())
+
+			// Get cluster class yaml. This is a hacky way to get cluster class with existing clusterctl api.
+			topologyYaml := clusterctl.ConfigCluster(ctx, clusterctl.ConfigClusterInput{
+				LogFolder:                filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
+				ClusterctlConfigPath:     clusterctlConfigPath,
+				KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+				InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+				Flavor:                   "topology",
+				Namespace:                namespace.Name,
+				ClusterName:              clusterName,
+				KubernetesVersion:        e2eConfig.GetVariable(capi_e2e.KubernetesVersion),
+				ControlPlaneMachineCount: pointer.Int64Ptr(1),
+				WorkerMachineCount:       pointer.Int64Ptr(1),
+			})
+			Expect(topologyYaml).To(Not(BeEmpty()))
+
+			// Create the cluster class.
```

Comment:
if we are to reuse this later across multiple workload clusters in the e2e test and it's not meant to be a 1-to-1 pattern, it might make more sense to extract this part outside of the `It` block and instead do this as part of the `BeforeSuite`

Reply:
Makes sense, I'll move it there.

Reply:
This is not possible right now because a cluster can use a topology only within its namespace, and since we create a namespace for each test, this will not work unless the topology is created in that namespace. There is an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/5673) open for this to make topology a namespaced name.

Reply:
Please see https://github.com/kubernetes-sigs/cluster-api/issues/5673#issuecomment-1124539439